benchmark/benchmark/bench_pca.py

```diff
@@ -0,0 +1,258 @@
+from benchmark.utils import with_benchmark
+
+
+class BenchmarkPCA(BenchmarkBase):
+    test_cls = PCA
+    unsupported_params = test_cls._param_excludes() + [
+        "featuresCol",
+        "labelCol",
+        "predictionCol",
+        "probabilityCol",
+        "rawPredictionCol",
+        "weightCol",
+        "leafCol",
+    ]
+
+    def add_arguments(self):
+        self._parser.add_argument("--no_cache", action='store_true', default=False, help='whether to enable dataframe repartition, cache and cout outside sparkcuml fit')
+
+    def score(self,
+        pc_vectors: np.ndarray,
+        transformed_df: DataFrame,
+        transformed_col: str
+    ) -> Tuple[float, float]:
+        """Computes a measure of orthonormality of the pc_vectors: maximum absolute deviation from 1 from all norms
+        and absolute deviation from 0 of all dot products between the pc_vectors.  This should be very small.
+
+        Also computes the sum of squares of the transformed_col vectors.  The larger the better.
+
+        PCA projection should have been performed on mean removed input for the second metric to be relevant.
+
+        Parameters
+        ----------
+        pc_vectors
+            principal component vectors.
+            pc_vectors.shape assumed to be (dim, k).
+        transformed_df
+            PCAModel transformed data.
+        transformed_col
+            Name of column with the PCA transformed data.
+            Note: This column is expected to be of pyspark sql 'array' type.
+
+        Returns
+        -------
+        Tuple[float, float]
+            The components of the returned tuple are respectively the orthonormality score and
+            the sum of squares of the transformed vectors.
+
+        """
+
+        pc_vectors = np.array(pc_vectors, dtype=np.float64)
+        k = pc_vectors.shape[1]
+        pc_vectors_self_prod = np.matmul(np.transpose(pc_vectors), pc_vectors)
+        orthonormality_score = np.max(np.abs(np.eye(k) - pc_vectors_self_prod))
+
+        def partition_score_udf(pdf_iter: Iterator[pd.DataFrame]) -> Iterator[float]:
+            partition_score = 0.0
+            for pdf in pdf_iter:
+                transformed_vecs = np.array(list(pdf[transformed_col]), dtype=np.float64)
+                partition_score += np.sum(transformed_vecs**2)
+            yield pd.DataFrame({'partition_score': [partition_score]})
+
+        total_score = (
+            transformed_df.mapInPandas(partition_score_udf,
+                                    StructType([StructField('partition_score',DoubleType(),True)]))
+                        .agg(sum('partition_score').alias('total_score'))
+                        .toPandas()
+        )
+        total_score = total_score['total_score'][0]
+        return orthonormality_score, total_score
+
+    def run_once(
+        self,
+        spark: SparkSession,
+        df: DataFrame,
+        features_col: Union[str, List[str]],
+        label_name: Optional[str],
+    ) -> None:
```

Comment:
Seems a typo. Use Dict[str, Any] as the return type instead of None?

Reply:
Good catch (turned out these files weren't being type checked until I moved them into the subfolder... so fixed that as well.